podvm-builder/azure-podvm-image-handler.sh

```diff
@@ -0,0 +1,656 @@
+        --gallery-name "${IMAGE_GALLERY_NAME}" \
+        --gallery-image-definition "${IMAGE_DEFINITION_NAME}"
+
+    return_code=$?
+
+    # Create Azure image definition if it doesn't exist
+
+    if [[ "${return_code}" == "0" ]]; then
+        echo "Image definition ${IMAGE_DEFINITION_NAME} already exists. Skipping creating the image definition"
+        return
+    fi
+
+    if [[ "${CONFIDENTIAL_COMPUTE_ENABLED}" == "yes" ]]; then
+        # Create the image definition. Add ConfidentialVmSupported feature
+        az sig image-definition create --resource-group "${AZURE_RESOURCE_GROUP}" \
+            --gallery-name "${IMAGE_GALLERY_NAME}" \
+            --gallery-image-definition "${IMAGE_DEFINITION_NAME}" \
+            --publisher "${IMAGE_DEFINITION_PUBLISHER}" \
+            --offer "${IMAGE_DEFINITION_OFFER}" \
+            --sku "${IMAGE_DEFINITION_SKU}" \
+            --os-type "${IMAGE_DEFINITION_OS_TYPE}" \
+            --os-state "${IMAGE_DEFINITION_OS_STATE}" \
+            --hyper-v-generation "${IMAGE_DEFINITION_VM_GENERATION}" \
+            --location "${AZURE_REGION}" \
+            --architecture "${IMAGE_DEFINITION_ARCHITECTURE}" \
+            --features SecurityType=ConfidentialVmSupported ||
+            error_exit "Failed to create Azure image definition"
+
+    else
+        az sig image-definition create --resource-group "${AZURE_RESOURCE_GROUP}" \
+            --gallery-name "${IMAGE_GALLERY_NAME}" \
+            --gallery-image-definition "${IMAGE_DEFINITION_NAME}" \
+            --publisher "${IMAGE_DEFINITION_PUBLISHER}" \
+            --offer "${IMAGE_DEFINITION_OFFER}" \
+            --sku "${IMAGE_DEFINITION_SKU}" \
+            --os-type "${IMAGE_DEFINITION_OS_TYPE}" \
+            --os-state "${IMAGE_DEFINITION_OS_STATE}" \
+            --hyper-v-generation "${IMAGE_DEFINITION_VM_GENERATION}" \
+            --location "${AZURE_REGION}" \
+            --architecture "${IMAGE_DEFINITION_ARCHITECTURE}" ||
+            error_exit "Failed to create Azure image definition"
+    fi
+
+    echo "Azure image definition created successfully"
+}
+
+# Function to use packer to create Azure image
+
+function create_image_using_packer() {
+    echo "Creating Azure image using packer"
+
+    # If any error occurs, exit the script with an error message
+    # The variables are set before calling the function
+
+    # Set the image version
+    # It should follow the Major(int).Minor(int).Patch(int)
+    IMAGE_VERSION="${IMAGE_VERSION_MAJ_MIN}.$(date +'%Y%m%d%S')"
+    export IMAGE_VERSION
+
+    # Set the image name
+    IMAGE_NAME="${IMAGE_BASE_NAME}-${IMAGE_VERSION}"
+    export IMAGE_NAME
+
+    # Set the base image details
+
+    if [[ "${PODVM_DISTRO}" == "rhel" ]]; then
+        export BASE_IMAGE_PUBLISHER="redhat"
+        export BASE_IMAGE_OFFER="rhel-raw"
+
+        # If CONFIDENTIAL_COMPUTE_ENABLED is set to yes, then set BASE_IMAGE_SKU to 9_3_cvm_sev_snp
+        # and IMAGE_DEFINITION_VM_GENERATION to V2
```

Comment:
IMAGE_DEFINITION_VM_GENERATION is always V2 isn't it? I don't see where it's forcefully changed if CoCo is used, also, IDK if overriding explicit values user sets is good idea, maybe we should just emit error if things aren't right? or log we are overriding it?

Reply:
Yes default is V2. The code doesn't change that.